src/test/java/com/springbootWithJPA/springBootRestExample/EmployeeControllerTest.java

```diff
@@ -0,0 +1,97 @@
+
+import static org.hamcrest.Matchers.hasSize;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(value = EmployeeController.class)
+public class EmployeeControllerTest {
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private EmployeeService employeeService;
+    @MockBean
+    private EmployeeRepository employeeRepository;
+
+    @InjectMocks
+    private EmployeeController controller;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Test
+    public void createEmployeeTest() throws Exception {
+      when(employeeService.createEmployee(Mockito.any(Employee.class))).thenReturn(getEmployees().get(0));
+        mockMvc.perform(post("/employee").contentType(MediaType.APPLICATION_JSON_VALUE).content(this.mapper.writeValueAsString(getEmployees().get(0)))).andExpect(status().isOk()).andExpect(jsonPath("$.firstName").value(getEmployees().get(0).getFirstName()));
+    }
+
+
+
+    @Test
+    public void testGetEmployee() throws Exception {
+        Mockito.when(employeeService.getEmployee(ArgumentMatchers.anyInt())).thenReturn((ResponseEntity.status(HttpStatus.OK).body(getEmployees().get(0))));
+        mockMvc.perform(get("/employee/1").contentType(MediaType.APPLICATION_JSON)).andExpect(status().isOk()).andExpect(jsonPath("$.firstName").value(getEmployees().get(0).getFirstName())).
+                andExpect(jsonPath("$.lastName").value(getEmployees().get(0).getLastName())).andExpect(jsonPath("$.emailId").value(getEmployees().get(0).getEmailId()));
+    }
+
+    @Test
+    public void getAllEmployeesTest() throws Exception {
+        Mockito.when(employeeService.getAllEmployees()).thenReturn(getEmployees());
+        mockMvc.perform(get("/employee/employees").contentType(MediaType.APPLICATION_JSON)).andExpect(status().isOk()).andExpect(jsonPath("$", hasSize(2))).andExpect(jsonPath("$[0].firstName").value(getEmployees().get(0).getFirstName())).
+                andExpect(jsonPath("$[0].lastName").value(getEmployees().get(0).getLastName())).andExpect(jsonPath("$[0].emailId").value(getEmployees().get(0).getEmailId())).andExpect(jsonPath("$[1].firstName").value(getEmployees().get(1).getFirstName())).
+                andExpect(jsonPath("$[1].lastName").value(getEmployees().get(1).getLastName())).andExpect(jsonPath("$[1].emailId").value(getEmployees().get(1).getEmailId()));
+    }
+
+    @Test
+    public void updateEmployeeTest() throws Exception {
+        when(employeeService.updateEmployee(1,getEmployees().get(0))).thenReturn(ResponseEntity.status(HttpStatus.OK).body(getEmployees().get(0)));
+    }
+
+    @Test
+    public void deleteEmployeeTest() throws Exception {
+       // mockMvc.perform(delete("/employee/2")).andExpect(status().isOk());
```

Comment:
Remove all commented code

Reply:
addressed in https://github.com/Jashika/springBootRestExample/pull/2/commits/0d318196ef3325c1a36a86cd1cdc1a38d93c5ccd